File: server/datasets/python_source-code/Multiple-G1/renamed_43_bst.py
class Node_7:
    def __init__(self, val_7):
        self.val_7 = val_7
        self.lhs_7 = None
        self.rhs_7 = None

class Tree_7:
    def __init__(self):
        self.rt_7 = None

    def ins_7(self, val_7):
        if self.rt_7 is None:
            self.rt_7 = Node_7(val_7)
        else:
            self._ins_7_rec(self.rt_7, val_7)

    def _ins_7_rec(self, nd_7, val_7):
        if val_7 < nd_7.val_7:
            if nd_7.lhs_7 is None:
                nd_7.lhs_7 = Node_7(val_7)
            else:
                self._ins_7_rec(nd_7.lhs_7, val_7)
        elif val_7 > nd_7.val_7:
            if nd_7.rhs_7 is None:
                nd_7.rhs_7 = Node_7(val_7)
            else:
                self._ins_7_rec(nd_7.rhs_7, val_7)

    def srch_7(self, val_7):
        return self._srch_7_rec(self.rt_7, val_7)

    def _srch_7_rec(self, nd_7, val_7):
        if nd_7 is None or nd_7.val_7 == val_7:
            return nd_7
        if val_7 < nd_7.val_7:
            return self._srch_7_rec(nd_7.lhs_7, val_7)
        return self._srch_7_rec(nd_7.rhs_7, val_7)

File: server/datasets/controlled_benchmarks/java/multi_student_cohorts/cohort_g1/renamed_43_bst.java
class Node_7 {
    int val_7;
    Node_7 lhs_7, rhs_7;
    public Node_7(int item) {
        val_7 = item;
        lhs_7 = rhs_7 = null;
    }
}
public class Tree_7 {
    public static void main(String[] args) {
        Tree_7 bst = new Tree_7();
        bst.ins_7(50);
        bst.srch_7(50);
    }
    Node_7 rt_7;
    public void ins_7(int val_7) {
        rt_7 = in_rec_7(rt_7, val_7);
    }
    Node_7 in_rec_7(Node_7 rt_7, int val_7) {
        if (rt_7 == null) return new Node_7(val_7);
        if (val_7 < rt_7.val_7) rt_7.lhs_7 = in_rec_7(rt_7.lhs_7, val_7);
        else if (val_7 > rt_7.val_7) rt_7.rhs_7 = in_rec_7(rt_7.rhs_7, val_7);
        return rt_7;
    }
    Node_7 srch_7(int val_7) {
        return sr_rec_7(rt_7, val_7);
    }
    Node_7 sr_rec_7(Node_7 rt_7, int val_7) {
        if (rt_7 == null || rt_7.val_7 == val_7) return rt_7;
        if (rt_7.val_7 < val_7) return sr_rec_7(rt_7.rhs_7, val_7);
        return sr_rec_7(rt_7.lhs_7, val_7);
    }
}